CMakeLists.txt

```diff
@@ -1,230 +1,164 @@
+target_include_directories(
+    ${target_name} ${warning_guard} PUBLIC
+    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+)
 
-  export(
-    EXPORT assert_targets
-    FILE ${CMAKE_CURRENT_BINARY_DIR}/assert/assert_targets.cmake
-    NAMESPACE assert::
-  )
+# require C++17 support
+target_compile_features(
+    ${target_name}
+    PUBLIC cxx_std_17
+)
 
-  configure_package_config_file(
-    cmake/assert-config.cmake.in
-    ${CMAKE_CURRENT_BINARY_DIR}/assert/assert-config.cmake
-    INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/assert
-  )
+# add pre-processor definitions corresponding to CMake options
+# fixme: supposedly generator expressions should work for this
+if(ASSERT_USE_MAGIC_ENUM)
+    target_compile_definitions(
+        ${target_name} PUBLIC
+        # $<$<BOOL:"${ASSERT_USE_MAGIC_ENUM}">:ASSERT_USE_MAGIC_ENUM>
+        ASSERT_USE_MAGIC_ENUM
+    )
+endif()
 
-  write_basic_package_version_file(
-    ${CMAKE_CURRENT_BINARY_DIR}/assert/assert-config-version.cmake
-    VERSION ${PACKAGE_VERSION}
-    COMPATIBILITY SameMajorVersion
-  )
 
-  install(
-    EXPORT assert_targets
-    FILE assert_targets.cmake
-    NAMESPACE assert::
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/assert
-  )
+# ---- Install Rules ----
 
-  install(
-    FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/assert/assert-config.cmake
-    ${CMAKE_CURRENT_BINARY_DIR}/assert/assert-config-version.cmake
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/assert
-  )
+if(NOT CMAKE_SKIP_INSTALL_RULES)
+    include(cmake/InstallRules.cmake)
 endif()
 
-# Don't run tests when library is used with add_subdirectory
-if(PROJECT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
-  if(WIN32)
-    add_custom_command(
-      TARGET assert POST_BUILD
-      COMMAND ${CMAKE_COMMAND} -E copy_if_different
-      $<TARGET_FILE:${ASSERT_CPPTRACE_TARGET_NAME}>
-      $<TARGET_FILE_DIR:assert>
-    )
-  endif()
-  if(ASSERT_BUILD_TESTS)
-    include(CTest)
-
-    set(
-      demo_sources
-      tests/demo/bar.cpp
-      tests/demo/baz/demo.cpp
-      tests/demo/demo.cpp
-      tests/demo/foo.cpp
-    )
-    add_executable(demo ${demo_sources})
-    target_link_libraries(demo PRIVATE assert)
-    target_compile_options(
-      demo
-      PRIVATE
-      "-DASSERT_USE_MAGIC_ENUM"
-      "-DASSERT_FAIL=custom_fail"
-      "-DASSERT_LOWERCASE"
-    )
 
-    add_executable(integration tests/integration/integration.cpp tests/integration/a.cpp tests/integration/x/a.cpp)
-    target_link_libraries(integration PRIVATE assert)
-    target_compile_options(
-      integration
-      PRIVATE
-      "-DASSERT_USE_MAGIC_ENUM"
-      "-DASSERT_FAIL=custom_fail"
-      "-DASSERT_LOWERCASE"
-    )
-    add_test(
-      NAME integration
-      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests
-      COMMAND
-      python3 run-tests.py $<TARGET_FILE:integration> ${CMAKE_BUILD_TYPE} ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_STANDARD}
-    )
+# ---- Setup Tests ----
+
+if(ASSERT_BUILD_TESTING)
```

Comment:
If this flag is being changed the ci scripts will need to be updated. I can take care of that.